File: app/src/main/java/space/model/ColonizerIndustry.java
package space.model;

public class ColonizerIndustry extends ShippingIndustry {
	
	public ColonizerIndustry(ShipCollection allies, PilotCollection pilots) {
		super(allies, pilots);
	}

	protected Ship createShip(Position planet) {
		
		timeToCompletion = 30.0f;
		
		return new Colonizer(planet);
	}
}

File: src/game/model/colonizer-industry.ts
import { GameConstants } from '../game-constants.ts';
import { Colonizer } from './colonizer.ts';
import { ShippingIndustry } from './shipping-industry.ts';
import { Position } from './position.ts';
import { Ship } from './ship.ts';

export class ColonizerIndustry extends ShippingIndustry {
  getMaterialCost() { return GameConstants.ColonizerIndustry.MaterialCost; }
  protected createShip(position: Position): Ship {
    this.productionDuration = GameConstants.ColonizerIndustry.ProductionDuration;
    this.timeToCompletion = this.productionDuration;
    return new Colonizer(position, this.allies);
  }
}
